src/server/pfs/cmds/cmds.go

```diff
@@ -535,12 +558,41 @@ $ {{alias}} test@master --new`,
 			}
 			defer c.Close()
 
+			var fromCommit *pfs.Commit
 			if newCommits && from != "" {
 				return errors.Errorf("--new and --from cannot be used together")
+			} else if newCommits || from != "" {
+				fromCommit = branch.NewCommit(from)
```

Comment:
I was going to comment that this was confusing in the case that branch isn't specified, but, as we discussed, `from` seems somewhat broken, so I'm not sure it matters